packages/scandipwa/src/route/Checkout/Checkout.container.js

```diff
@@ -283,6 +355,32 @@ export class CheckoutContainer extends PureComponent {
         toggleBreadcrumbs(true);
     }
 
+    saveShippingFieldsAsShippingAddress(address, is_virtual) {
+        const {
+            street_0,
+            street_1,
+            street_2,
+            shipping_method,
+            ...data
+        } = address;
+        const { savedEmail } = this.props;
+        const { checkoutStep } = this.state;
+
+        const checkoutData = (
+            is_virtual
+                ? { shippingAddress: {}, isGuestEmailSaved: false }
+                : {
+                    shippingAddress: data,
+                    isGuestEmailSaved: savedEmail && checkoutStep !== SHIPPING_STEP
+                }
+        );
+
+        this.setState({
+            email: savedEmail,
+            ...checkoutData
```

Comment:
If `checkoutData` will have an `email` field, it will override the one from the top.
Please spread objects first, put additional properties and overrides next.

Reply:
`checkoutData` doesn't have `email`,
it only has `shippingAddress` and `isGuestEmailSaved`

But cool, ok, to follow a design pattern